src/libstd/thread/mod.rs

```diff
@@ -1036,15 +1037,15 @@ pub fn park_timeout(dur: Duration) {
 /// [`Thread`]: ../../std/thread/struct.Thread.html
 #[stable(feature = "thread_id", since = "1.19.0")]
 #[derive(Eq, PartialEq, Clone, Copy, Hash, Debug)]
-pub struct ThreadId(u64);
+pub struct ThreadId(NonZeroU64);
 
 impl ThreadId {
     // Generate a new unique thread ID.
     fn new() -> ThreadId {
         // We never call `GUARD.init()`, so it is UB to attempt to
         // acquire this mutex reentrantly!
         static GUARD: mutex::Mutex = mutex::Mutex::new();
```

Comment:
Out of interest, is there any reason we don't use an atomicu64 here instead?

Reply:
Using a mutex is slightly more portable than an `AtomicU64`, and this isn't really perf critical so we didn't need to reach for that.

Reply:
@alexcrichton: At @Structure-Systems we're logging thread IDs at a high frequency. How will this choice impact performance?

Reply:
@sanmai-NL have you profiled and seen this as a problem?

Reply:
This mutex is only locked when a thread is created, not when its thread id is read, FYI.